src/django_components/component.py

```diff
@@ -254,33 +258,76 @@ def get_template(self, context: Mapping) -> Template:
             f"Note: this attribute is not required if you are overriding the class's `get_template*()` methods."
         )
 
+    def render_from_input(self, context: Context, args: Union[List, Tuple], kwargs: Dict) -> str:
```

Comment:
Created this function so that the logic that manages / calls `get_context_data` lives on the `Component` class. The reason for this is also that, down the line, I'd like a feature to render a `Component` class directly, without needing to define a Template, but still being able to pass in args/kwargs like in a template. And so with this function, we can do so.